python/cudf/dataframe/dataframe.py

```diff
@@ -353,6 +353,28 @@ def __array_ufunc__(self, ufunc, method, *inputs, **kwargs):
         else:
             return NotImplemented
 
+    def __array_function__(self, func, types, args, kwargs):
+
+        cudf_df_module = DataFrame
+        for submodule in func.__module__.split(".")[1:]:
+            # point cudf to the correct submodule
+            if hasattr(cudf_df_module, submodule):
+                cudf_df_module = getattr(cudf_df_module, submodule)
+            else:
+                return NotImplemented
+
```

Comment:
Might want to add a check for type here like in cupy: https://github.com/cupy/cupy/blob/master/cupy/core/core.pyx#L1303-L1306

Was not sure if we only wanted to only limit np functions with cudf_df/cudf_series  as arguments. (for arguments that have array_function like np/cupy/dask arrays etc) 
```python
handled_types = [cudf_df_module,cudf_series_module]
for t in types:
   if t not in handled_types:
     return NotImplemented
```

Reply:
Yes we should limit it for now at least.

Reply:
Fixed with https://github.com/rapidsai/cudf/pull/2157/commits/303d31da2f68d4147f1e17c3f91c2f90b310d2e1 .  

Sorry for the rebase but with the `dask-cudf` I felt it was the cleanest way to resolve conflicts.